batch2/batch/worker.py

```diff
@@ -714,6 +750,32 @@ def __init__(self):
                 del self.jobs[job.id]
                 self.last_updated = time.time()
 
+    async def post_job_timing(self, job):
+        body = {
+            'status': await job.status()
+        }
+
+        delay = 0.1
+        while True:
+            try:
+                async with aiohttp.ClientSession(
+                        raise_for_status=True, timeout=aiohttp.ClientTimeout(total=60)) as session:
+                    await session.post(
+                        deploy_config.url('batch2-driver', '/api/v1alpha/instances/job_timing'),
+                        json=body, headers=self.headers)
+                    return
+            except asyncio.CancelledError:  # pylint: disable=try-except-raise
+                raise
+            except Exception as e:
+                if isinstance(e, aiohttp.ClientResponseError) and e.status == 404:   # pylint: disable=no-member
+                    raise
+                log.exception(f'failed to post {job} timing, retrying')
+
+            # exponentially back off, up to (expected) max of 30s
+            t = delay * random.uniform(0.7, 1.3)
```

Comment:
Hmm, is it necessary to infinitely retry job timings, even for errors?  I'm not sure.  I wonder if retry_transient_errors woudl be suitable here.

Reply:
I'm not sure what the right answer is. For the job_started, it doesn't matter so much, but if a job is cancelled we definitely want that information of how long it had been running.

Reply:
So with my other suggestions, I'd be OK making this just retry transients.  The only case we'd lose information is when a job starts, this fails, and then the job is cancelled or preempted.  That seems rare enough to be OK.  And if this fails, that probably means something much more serious is wrong.